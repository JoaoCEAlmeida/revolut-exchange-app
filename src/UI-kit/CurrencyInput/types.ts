export interface ComponentProps {
  pickerProps?: object;
  inputProps?: object;
}
