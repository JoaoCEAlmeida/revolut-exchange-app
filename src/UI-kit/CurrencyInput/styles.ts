import styled from 'styled-components';

export const CurrencyInputWrapper = styled.div`
  display: flex;
  justify-content: space-between;

  padding: 20px;
  background-color: ${(props) => props.theme.colors.secondary.main};
  border-radius: 7px;
  margin-top: 5px;
  margin-bottom: 5px;
  &:focus-within {
    background-color: ${(props) => props.theme.colors.secondary.contrast};
  }
`;
