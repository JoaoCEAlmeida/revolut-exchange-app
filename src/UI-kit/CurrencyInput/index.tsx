import React from 'react';
import { CurrencyInputWrapper } from './styles';

import CurrencyPicker from '../CurrencyPicker';

import Input from '../Input';

import { ComponentProps } from './types';

const CurrencyInput: React.FC<ComponentProps> = ({ pickerProps, inputProps }) => {
  return (
    <CurrencyInputWrapper>
      <CurrencyPicker {...pickerProps} />
      <Input placeholder="0" {...inputProps} />
    </CurrencyInputWrapper>
  );
};

export default CurrencyInput;
