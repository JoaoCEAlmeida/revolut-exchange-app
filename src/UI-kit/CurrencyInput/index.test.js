import { screen, fireEvent } from '@testing-library/react';
import CurrencyInput from './index';
import renderWithProviders from '../../test/providers';

const defaultProps = {
  inputProps: {
    placeholder: 'Placeholder',
    type: 'button',
    error: '',
    onChange: jest.fn()
  },
  pickerProps: {
    currency: 'EUR',
    balance: 1000,
    onClick: jest.fn()
  }
};

describe('CurrencyInput Tests', () => {
  test('should render correctly', () => {
    const { CurrencyInputTest } = renderWithProviders(() => <CurrencyInput {...defaultProps} />);

    expect(CurrencyInputTest).toMatchSnapshot();
  });

  test('should allow editing', () => {
    renderWithProviders(() => <CurrencyInput {...defaultProps} />);
    const input = screen.getByPlaceholderText('Placeholder');

    expect(input.value).toBe('');

    fireEvent.change(input, { target: { value: '1000' } });

    expect(input.value).toBe('1000');
  });

  test('should fire the onClick event', () => {
    const clickHandler = jest.fn();

    renderWithProviders(() => (
      <CurrencyInput
        {...defaultProps}
        pickerProps={{ ...defaultProps.pickerProps, onClick: clickHandler }}
      />
    ));

    fireEvent.click(screen.getByText('EUR'));

    expect(clickHandler).toHaveBeenCalled();
  });

  test('should have the correct balance', () => {
    const balance = 12.23;
    renderWithProviders(() => (
      <CurrencyInput
        {...defaultProps}
        pickerProps={{ ...defaultProps.pickerProps, balance: balance }}
      />
    ));

    expect(screen.getByTestId('test-balance').textContent).toEqual(`Balance: ${balance}`);
  });

  test('should have the correct currency', () => {
    const currency = 'GBP';
    renderWithProviders(() => (
      <CurrencyInput
        {...defaultProps}
        pickerProps={{ ...defaultProps.pickerProps, currency: currency }}
      />
    ));
    expect(screen.getByTestId('test-currency').textContent).toEqual(`${currency}`);
  });
});
