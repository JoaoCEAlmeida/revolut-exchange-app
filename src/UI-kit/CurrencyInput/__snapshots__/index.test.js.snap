// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInput Tests should render correctly 1`] = `undefined`;

exports[`Input Tests should render correctly 1`] = `undefined`;
